strongbox-web-core/src/test/java/org/carlspring/strongbox/rest/common/RestAssuredBaseTest.java

```diff
@@ -0,0 +1,255 @@
+    {
+        return getArtifactAsStream(url, -1);
+    }
+
+    protected InputStream getArtifactAsStream(String url,
+                                              int offset)
+    {
+        return new ByteArrayInputStream(getArtifactAsByteArray(url, offset));
+    }
+
+    /**
+     * Converts response output to byte array to properly use it later as a stream.
+     * RestAssured-specific case for working with file uploading when multipart specification is not used.
+     */
+    protected byte[] getArtifactAsByteArray(String url,
+                                            int offset)
+    {
+        MockMvcRequestSpecification o = given().contentType(MediaType.TEXT_PLAIN_VALUE);
+        int statusCode = 200;
+        if (offset != -1)
+        {
+            o = o.header("Range", "bytes=" + offset + "-");
+            statusCode = 206;
+        }
+
+        logger.debug("[getArtifactAsByteArray] URL " + url);
+
+        MockMvcResponse response = o.when().get(url);
+        Headers allHeaders = response.getHeaders();
+        logger.debug("HTTP GET " + url);
+        logger.debug("Response headers:");
+
+        allHeaders.forEach(header -> logger.trace("\t" + header.getName() + " = " + header.getValue()));
+
+        response.then().statusCode(statusCode);
+        byte[] result = response.getMockHttpServletResponse().getContentAsByteArray();
+
+        logger.debug("Received " + result.length + " bytes.");
+
+        return result;
+    }
+
+    protected void copy(String path,
+                        String srcStorageId,
+                        String srcRepositoryId,
+                        String destStorageId,
+                        String destRepositoryId)
+    {
+        given()
+                .contentType(MediaType.TEXT_PLAIN_VALUE)
+                .params("path", path,
+                        "srcStorageId", srcStorageId,
+                        "srcRepositoryId", srcRepositoryId,
+                        "destStorageId", destStorageId,
+                        "destRepositoryId", destRepositoryId)
+                .when()
+                .post(getContextBaseUrl() + "/storages/copy")
+                .peek()
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .extract();
+    }
+
+    protected void delete(String storageId,
+                          String repositoryId,
+                          String path)
+            throws ArtifactOperationException
+    {
+        delete(storageId, repositoryId, path, false);
+    }
+
+    protected void delete(String storageId,
+                          String repositoryId,
+                          String path,
+                          boolean force)
+            throws ArtifactOperationException
+    {
+        String url = getContextBaseUrl() + "/storages/" + storageId + "/" + repositoryId;
+
+        given().contentType(MediaType.TEXT_PLAIN_VALUE)
+               .param("path", path)
```

Comment:
@AlexOreshkevich,

Unless I'm mistaken, this variable needs to go, as it's all path-variable based, and not using the request parameters.